torch/_tensor_str.py

```diff
@@ -266,6 +266,20 @@ def _str(self):
         if values.numel() == 0:
             values_str += ', size=' + str(tuple(values.shape))
         tensor_str = indices_prefix + indices_str + '),\n' + ' ' * indent + values_prefix + values_str + ')'
+    elif self.is_quantized:
+        suffixes.append('size=' + str(tuple(self.shape)))
+        suffixes.append('dtype=' + str(self.dtype))
+        suffixes.append('quantization_scheme=' + 'per_tensor_affine')
+        suffixes.append('scale=' + str(self.q_scale()))
+        suffixes.append('zero_point=' + str(self.q_zero_point()))
+        values_prefix = 'float_values=tensor('
```

Comment:
I don't really find the "print both" rationale convincing (outside of debugging, but there you can print whatever you want).

The way I think about this is as follows:
Ideally, the print should be as close to usable as possible (if you print a small strided tensor you can actually copy-paste the print and get the tensor out).  This doesn't work in all conditions (i.e. very large tensors), but it's a good guiding principle.

Now, the quantized APIs are still subject to change, but I think it's reasonable to say no one would ever pass both the float and integer values together.  Since the usual use case is quantizing a float tensor, I'd lean towards just printing the float values (and you don't have to label them `float_values`).

Now, maybe this is a little strange when you combine with the zero_point and scale, because those are already "included" in the float values, but I'd really view those as should be belonging to the dtype, which we always print in this case.

So in summary, if you don't feel strongly otherwise, I would suggest:
- not printing int_repr.
- not including float_values
- not printing scale/zero point as tensors, but just as values.

Reply:
Sure